src/dropdown/dropdown.scss

```diff
@@ -81,6 +81,17 @@
   color: var(--color-fg-default);
   text-overflow: ellipsis;
   white-space: nowrap;
+  cursor: pointer;
+
+  &[disabled] {
```

Comment:
For the `SelectMenu-item` we use  [`:disabled`](https://github.com/primer/css/blob/e8d5402a26f0e4428cb0cd0694569da3c2ed505e/src/select-menu/select-menu.scss#L404) so that it would still work for `disabled=false`.

And there is also the `[aria-disabled='true']` selector so you can manually set an element to disabled.

```suggestion
  &:disabled,
  [aria-disabled='true'] {
```

Reply:
Hmm interesting. I tried that and it didn't work. I have no idea why? Happy to add the `aria-disabled` selector tho.

Reply:
Ah ok, it doesn't work because these menu items are implemented as `<a>` tags which don't support being disabled.

Reply:
Ohhh.. I see. There is probably an accessibility reason why `<a>`'s shouldn't be disabled? [This article](https://www.scottohara.me/blog/2021/05/28/disabled-links.html) mentions that removing the `href` and adding `aria-disabled="true"` should be ok. /cc @primer/accessibility If so, then I guess we wouldn't need `[disabled]` nor `:disabled`.

```suggestion
  &[aria-disabled='true'] {
```

Reply:
I also tried to use a `<button disabled>` as a `dropdown-item`. But we would need additional styling to add support:

```html
<button class="dropdown-item" disabled>Open in GitHub Desktop</button>
```

![Screen Shot 2022-03-01 at 15 46 47](https://user-images.githubusercontent.com/378023/156119503-e9fcda9b-ccf5-4411-b907-070f99ebffda.png)

Reply:
I would avoid the concept of disabled links. The closest semantic approach to "disabled links" is just regular text.

A disabled button makes more sense in general but in the context of a menu or select it's probably not the best choice. Something like [option (role)](https://www.w3.org/TR/wai-aria-1.1/#option) is probably best and that does support `aria-disabled`. I would imagine that this kind of change requires much more work so may not be practical.

@simurai mentions [Disabling a link](https://www.scottohara.me/blog/2021/05/28/disabled-links.html) and I agree that this might be the most practical approach. That said, the last line of the article is worth noting:

> Use wisely, carefully, and hopefully rarely.

I'm finding it hard to make a concrete recommendation because the menu probably needs a lot more work to really enhance the accessibility of the component. See: 

- https://github.com/github/accessibility/issues/600
- https://github.com/github/accessibility/issues/581

Reply:
@benjiallen what would you think about rendering a button instead of a link for the purposes of showing it disabled?